in_toto/model.go

```diff
@@ -41,6 +81,25 @@ type Signature struct {
 	Sig   string `json:"sig"`
 }
 
+func validateSignature(signature Signature) error {
+	if err := validateHexString(signature.KeyId); err != nil {
+		return fmt.Errorf("keyid: %s", err.Error())
+	}
+	if err := validateHexString(signature.Sig); err != nil {
+		return fmt.Errorf("signature: %s", err.Error())
+	}
+	return nil
+}
```

Comment:
For `validateSignature` and also `validatePubKey` and `validateRSAPubKey` it might be worth to add the keyid to each error message, to make troubleshooting easier.

Reply:
The incorrect `keyid`? That's part of `err.Error()`.

```func validateHexString(str string) error {
	formatCheck, _ := regexp.MatchString("^[a-fA-F0-9]+$", str)
	if !formatCheck {
		return fmt.Errorf("'%s' is not a valid hex string", str)
	}
	return nil
}```

Reply:
Yes but only if the `KeyId` is the fail reason. What I'm suggesting is to add the `KeyId` to all error messages, where the KeyId helps to identify the piece of data, even if the `KeyId` is not the fail reason.

Reply:
Gotcha.